### PR TITLE
updated eks cluster create for nodepool API changes

### PR DESCRIPTION
**Pull request by asutoshpalai**
1. **What this PR does / why we need it**:

- Updates EKS Cluster creation API calls to create Nodepools separately. This change is needed to reflect the API changes on the TMC side.
- Updates EKS Nodepool schema to add support for Custom AMIs.

2. **Which issue(s) this PR fixes**

        Fixes OLYMP-36757

3. **Additional information**

4. **Special notes for your reviewer**

### CODE REVIEW

File: internal/models/ekscluster/spec.go [hunk 1]
```diff
@@ -25,9 +25,6 @@ type VmwareTanzuManageV1alpha1EksclusterSpec struct {
 	// EKS config for the cluster control plane.
 	Config *VmwareTanzuManageV1alpha1EksclusterControlPlaneConfig `json:"config,omitempty"`
 
-	// Nodepool definition for the cluster.
```
Comment (Vasundharashukla) [inline internal/models/ekscluster/spec.go:28]: Is the nodepool definition getting moved to some other place?

File: internal/resources/ekscluster/resource_nodepool.go [hunk 2]
```diff
@@ -64,10 +64,30 @@ var nodepoolSpecSchema = &schema.Schema{
...
+				Type:        schema.TypeList,
+				Description: "AMI info for the nodepool if AMI type is specified as CUSTOM",
+				Optional:    true,
+				MaxItems:    1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						amiIDKey: {
+							Type:        schema.TypeString,
+							Description: "ID of the AMI to be used",
+							Optional:    true,
```
Comment (Vasundharashukla) [inline internal/resources/ekscluster/resource_nodepool.go:81]: Do we have a default value for `amiID` and `overrideBootstrap`?

File: internal/models/ekscluster/spec.go (hunk 1 above)
Comment (asutoshpalai) [reply internal/models/ekscluster/spec.go:28]: Yes, we are removing it from the cluster's spec. All the nodepool operations to be done via nodepool's own APIs.

The field is still there in the API but is marked for deprecation, hence this MR.

File: internal/resources/ekscluster/resource_nodepool.go (hunk 2 above)
Comment (asutoshpalai) [reply internal/resources/ekscluster/resource_nodepool.go:81]: No, they are specific to the AMIs that the user wants to use.

**Comment by shreyassreenivas** [comment]
@asutoshpalai , have all the changes been manually validated and are the acceptance run ? 
can you please post the validation report for this MR.

Question : 
Now since, the nodepool is a seperate API, do you want to decouple the EKS resource in the provider too ? what is your take on this.

**Comment by asutoshpalai** [comment]
Validation of CRUD operations:

Screen Shot 2023-03-20 at 9 12 28 AM
Screen Shot 2023-03-20 at 9 20 34 AM
Screen Shot 2023-03-20 at 9 37 29 AM

**Comment by asutoshpalai** [comment]
> Now since, the nodepool is a seperate API, do you want to decouple the EKS resource in the provider too ? what is your take on this.

Yes, that's on the map. Since that will require slightly more intrusive changes and intensive testing, we want this to be done first. Currently, we want to unblock the API change to be merged and get ready for GA.

File: go.mod [hunk 3]
```diff
@@ -101,3 +101,18 @@ require (
 	sigs.k8s.io/structured-merge-diff/v4 v4.1.2 // indirect
 	sigs.k8s.io/yaml v1.2.0 // indirect
 )
+
+require (
```
[the remark this answers is not in the document]
Comment (shreyassreenivas) [reply go.mod:105]: can you address this require block

**Review by Vasundharashukla — APPROVED** [review]

**Review by shreyassreenivas — APPROVED** [review]

**Comment by shreyassreenivas** [comment]
@asutoshpalai .I hope the acceptance tests for the resource has been run and its passing. 
If yes, please provide the screenshots

**Pull request merged**